core/src/main/java/hudson/ExtensionList.java

```diff
@@ -350,12 +354,11 @@ public void refresh(ExtensionComponentSet delta) {
                 fireOnChangeListeners = true;
```

Comment:
Could simplify slightly by just `return true` here. The original reason for the `fireOnChangeListeners` var was to call `fireOnChangeListeners` outside the `synchronized` block, which is no longer relevant.